aptos-move/move-examples/marketplace/sources/token_offer.move

```diff
@@ -0,0 +1,788 @@
+address marketplace {
+/// Provides the ability to make token offers to both Tokenv1 and Tokenv2 tokens.
+/// A token offer allows an entity to place a bid on a token at any time. The amount
+/// offered is extracted from their account and stored at an escrow. A seller can then
+/// exchange the token for the escrowed payment. If it is a tokenv2 or the recipient
+/// has enabled direct deposit, the token is immediately transferred. If it is tokenv1
+/// without direct deposit, it is stored in a container until the recipient extracts it.
+module token_offer {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::String;
+
+    use aptos_framework::coin::{Self, Coin};
+    use aptos_framework::object::{Self, DeleteRef, Object};
+    use aptos_framework::timestamp;
+
+    use aptos_token::token as tokenv1;
+
+    use aptos_token_objects::royalty;
+    use aptos_token_objects::token::{Self as tokenv2, Token as TokenV2};
+
+    use marketplace::events;
+    use marketplace::fee_schedule::{Self, FeeSchedule};
+    use marketplace::listing::{Self, TokenV1Container};
+    use aptos_token::token::TokenId;
+
+    /// No token offer defined.
+    const ENO_TOKEN_OFFER: u64 = 1;
+    /// No coin offer defined.
+    const ENO_COIN_OFFER: u64 = 2;
+    /// This is not the owner of the token.
+    const ENOT_TOKEN_OWNER: u64 = 3;
+    /// This is not the owner of the token offer.
+    const ENOT_OWNER: u64 = 4;
+    /// The token offer has expired.
+    const EEXPIRED: u64 = 6;
+
+    // Core data structures
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Create a limed lifetime offer to buy a token. The token and
```

Comment:
Think there's a typo here 😛 supposed to be "limited or lifetime"? Not sure

Other than this lgtm!